src/framework/components/anim/property-locator.js

```diff
@@ -0,0 +1,65 @@
+Object.assign(pc, function () {
```

Comment:
I suggest you consider something like for encode and decode:

```
encode: function(locator) {
    return AnimBinder.joinPath(AnimBinder.joinPath(locator[0]), locator[1], AnimBinder.joinPath(locator[2]));
},

decode: function(locator) {
    var tmp = AnimBinder.splitPath(locator);
    return [AnimBinder.splitPath(tmp[0]), tmp[1], AnimBinder.splitPath(tmp[2]));
}
```

This is much simpler to understand and maintain (well for me at least :) ) and also it looks like the code below doesn't escape /'s correctly.

Reply:
Sure i'll rewrite this 👍